app/templates/buyers/brief_overview.html

```diff
@@ -20,12 +20,37 @@
           <form action="{{ url_for('buyers.delete_a_brief', framework_slug=framework.slug, lot_slug=brief.lotSlug, brief_id=brief.id) }}" method="POST">
             <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
             <input type="hidden" name="delete_confirmed" value="true" />
-            <div class="banner-destructive-with-action">
-              <p class="banner-message">
-               Are you sure you want to delete these requirements?
-              </p>
-              <button type="submit" class="button-destructive banner-action">Yes, delete</button>
-            </div>
+            {%
+              with
+              message = "Are you sure you want to delete these requirements?",
+              type = "destructive",
+              action = '<button type="submit" class="button-destructive banner-action">Yes, delete</button>'|safe
+            %}
+              {% include "toolkit/notification-banner.html" %}
+            {% endwith %}
+          </form>
+        </div>
+      {% endif %}
+      {% if withdraw_requested %}
```

Comment:
We should look at centralising these patterns in the frontend-toolkit given that flash messages are now generally handled there.

https://github.com/alphagov/digitalmarketplace-frontend-toolkit/blob/master/toolkit/templates/flash_messages.html

That doesn't currently support flashes with actions, but we should add it if we need it so that the work doesn't have to be done again when another app (inevitably) wants to do the same thing.

Reply:
I was wondering whether that should be in scope of this story. I made the (reluctant) choice to pass through `"<some html>"|safe` to the toolkit for now, as it's difficult to know where this pattern is used elsewhere and what would break with the change. Also, how much to include in the new pattern - just the action button text? CSRF tokens/POST URL? The whole form div?

Reply:
It's hard to justify pulling it into a story because it does expand the scope a little bit, but I think we need to try to take positive actions like this wherever possible.

Things like CSRF should be centralised as much as possible IMO; the fewer places you need to implement anything security-focused, the better. That said, I don't immediately know how we'd design the interface for the action flash. It's probably not a small thing. :)

Reply:
We do have a tech debt ticket (two, in fact) for this already https://trello.com/c/ZUmDyovj/23-flash-messages-dry-etc. I've added some details to that ticket that I mentioned above (CSRF etc).